alf/algorithms/td_loss.py

```diff
@@ -94,24 +96,33 @@ def forward(self, experience, value, target_value):
         Returns:
             LossInfo: with the ``extra`` field same as ``loss``.
         """
+        if experience.reward.ndim == 3:
+            gamma = self._gamma
+            if gamma.ndim == 0:  # scalar
+                gamma = gamma.unsqueeze(0)
+            discounts = torch.ger(experience.discount.reshape(-1), gamma)
+            discounts = discounts.reshape(experience.discount.shape + (-1, ))
```

Comment:
```python
# [T, B, d] or [T, B, 1]
discounts = discount.unsqueeze(-1) * gamma
```

Reply:
I didn't know [..., 1] * [D] is broadcastable.